packages/json-deserializer/src/JsonDeserializer.test.ts

```diff
@@ -0,0 +1,98 @@
+import { jsonDeserializer } from "./JsonDeserializer";
+import { createContext, createEvent } from "@lambda-middleware/utils";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { APIGatewayProxyObjectEvent } from "./types/APIGatewayProxyObjectEvent";
+
+describe("jsonDeserializer", () => {
+  let processedEvent: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>;
+
+  const dummyHandler = async (
+    event: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>
+  ) => {
+    processedEvent = event;
+    return {} as APIGatewayProxyResult;
+  };
+  const handlerWithMiddleware = jsonDeserializer()(dummyHandler);
+
+  const testObject = {
+    a: "json",
+    object: {
+      with: ["multiple", "types"],
+      ofProperties: {
+        numbers: 2345,
+        booleans: true,
+        stringValues: "testString",
+        nulls: null,
+        undefineds: undefined,
+        andArrays: ["ofStrings", 42, null],
+      },
+    },
+  };
+
+  it("should deserialize objects if the event has a json content-type header", async () => {
```

Comment:
Sure, in the past I've had people ask for all the tests to contain should in the past on other projects to reflect expected behaviour, rather than a statement of fact, happy to fit with convention though.